src/lib/tools/navigation/NavigationTool.ts

```
@@ -35,13 +35,44 @@ export class NavigationTool {
+      const lower = (url || '').toLowerCase()
+      // Simple contains match
+      if (failed.some(f => lower.includes(f))) return true
+      // Structured host/path matching
+      try {
+        const target = new URL(url)
+        for (const f of failed) {
+          try {
+            const fu = new URL(f)
+            if (target.hostname === fu.hostname) {
+              if (fu.pathname && target.pathname.includes(fu.pathname)) return true
+              const m = fu.pathname.match(/\/(?:pdf|abs)\/([^\/?#]+)/)
```

ESLint

[error] Unnecessary escape character: \/.

(no-useless-escape)